### ISSUE TITLE
Structured List with selection should be a table/grid

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#5065]
Screen reader/keyboard support for our StructuredList component is severely lacking as noted in #1937. In order to have a more usable experience for users with screen reader's the underlying markup for StructuredList should be changed to either a table or a grid as noted in [this comment](https://github.com/carbon-design-system/carbon/issues/1882#issuecomment-557181329).

This work has two main components (as noted in the above referenced comment):
- Update the markup either to table or grid 
- Figure out how to do row-level interactions like selection for this use-case

**Comment by asudoh** [comment]
@dakahn Is this issue fixed by #5066? Thanks!

**Comment by dakahn** [comment]
Only partly -- remaining fixes needed design consult and will be in a subsequent PR 🏄

**Comment by dakahn** [comment]
closed by https://github.com/carbon-design-system/carbon/pull/8172
